### PR TITLE
[NP-5040] - Check RW/RO property permissions for filters and columns

### PR DESCRIPTION
**Pull request by nanopay-arthur**
Ref: https://nanopay.atlassian.net/browse/NP-5040

### CODE REVIEW

File: src/foam/u2/filter/FilterView.js [hunk 1]
```diff
@@ -275,18 +237,30 @@ foam.CLASS({
...
+      value: false
     }
   ],
 
   methods: [
-    function render() {
+    function init() {
+      this.onDetach(this.searchColumns$.sub(this.updateFilters));
+      this.onDetach(this.dao$.sub(this.updateFilters));
+
```
Comment (sarthak-marwaha) [inline src/foam/u2/filter/FilterView.js]: Remove extra line

File: src/foam/u2/filter/FilterView.js [hunk 2]
```diff
@@ -473,7 +472,58 @@ foam.CLASS({
             continue;
         }
       }
+      this.mementoUpdated = true;
```
Comment (sarthak-marwaha) [inline src/foam/u2/filter/FilterView.js:477]: The slot won't be called if the value is always set to true, here we can do `this.memetoUpdated = ! this.mementoUpdated` that way the slot will always be called. We would also need to remove the if from the slot

**Review by sarthak-marwaha — APPROVED** [review]

File: src/foam/u2/filter/FilterView.js [hunk 3]
```diff
@@ -473,7 +474,59 @@ foam.CLASS({
...
+        this.columns = columns.filter(function(c) {
+        //  to account for nested columns like approver.legalName
+        if ( c.split('.').length > 1 ) return false;
+
+        var a = of.getAxiomByName(c);
+
+        if ( ! a ) console.warn("Column does not exist for " + of.name + ": " + c);
+
+        return a
+          && ! a.storageTransient
```
Comment (jlhughes) [inline src/foam/u2/filter/FilterView.js:513]: why filter out 'storageTransient'?  storageTransient may not be persisted to disk but can be indexed in MDAO and are marshalled by medusa.

Comment (nanopay-arthur) [reply src/foam/u2/filter/FilterView.js:513]: This is old logic, but I'm assuming it was added for consistency purposes. If the data is not actually stored server-side, we don't want it to be displayed in any way, even if it is stored in the MDAO.

**Review by jlhughes — APPROVED** [review]

**Pull request merged**